packages/headless/src/features/configuration/configuration-state.ts

```diff
@@ -55,6 +42,54 @@ export interface ConfigurationState {
    * The global headless engine Knowledge configuration.
    */
   knowledge: KnowledgeState;
+  /**
+   * The environment in which the Coveo cloud organization is hosted.
+   *
+   * The `dev` and `stg` environments are only available internally for Coveo employees (e.g., Professional Services).
+   *
+   * Defaults to `prod`.
+   */
+  environment: PlatformEnvironment;
+}
+
+export interface CommerceState {
+  /**
+   * The Commerce API base URL to use.
+   *
+   * By default, will append `/rest/organizations/{organizationId}/commerce/v2` to the automatically resolved
+   * `platformUrl` value.
+   *
+   * If necessary, you can override this value by specifying a `proxyBaseUrl` in the configuration of your commerce
+   * engine, or in the payload of an `updateBasicCommerceConfiguration` action.
+   */
+  apiBaseUrl: string;
+}
```

Comment:
Is there a case where the base configuration of a headless engine not be a Commerce Engine, but also still needs the `apiBaseUrl` of the Commerce API?

Reply:
No, I agree with you that it would be better to have a commerce-specific configuration state.